spring-session/src/main/java/org/springframework/session/data/gemfire/AbstractGemFireOperationsSessionRepository.java

```diff
@@ -495,18 +503,8 @@ public synchronized void setPrincipalName(String principalName) {
 
 		/* (non-Javadoc) */
 		public synchronized String getPrincipalName() {
-			String principalName = getAttribute(PRINCIPAL_NAME_INDEX_NAME);
-
-			if (principalName == null) {
-				Object authentication = getAttribute(SPRING_SECURITY_CONTEXT);
-
-				if (authentication != null) {
-					Expression expression = this.parser.parseExpression("authentication?.name");
-					principalName = expression.getValue(authentication, String.class);
-				}
-			}
-
-			return principalName;
+			Map<String, String> indexes = indexResolver.resolveIndexesFor(this);
```

Comment:
This would benefit from `IndexResolver` having a method to resolve single index. However the GemFire repository implementation is the one I'm least familiar with, so this altogether could perhaps be refactored in a more optimal way.